### ISSUE TITLE
Jump to previous/next matching indentation point?

### ISSUE THREAD

**Issue by superDuperCyberTechno** [helix-editor/helix#9819]
It's fairly often that I need to refer to the current scope of my work. This could easily be traversed if Helix had a way to jump to where the current indentation is started/ended. Any further movement from this start/end-point would jump the cursor to the next indentation point.

Example: 
1. Jumping to the previous matching indent point from `logic4` will place the cursor at the starting character of `indent2`. 
2. Repeating the last movement would place the cursor at the starting character of `indent1`. 
3. Jumping to the next matching indent point from `indent1` would jump the cursor to `outdent1`.

```
indent1
    logic1
    logic2
    indent2
        logic3
        logic4
    outdent2
outdent1
```

**Comment by 7ombie** [comment]
I was surprised that No. 3 wasn't supported already. Being able to jump from any of the first four characters of `indent1` to the first four characters of `outdent1` (and *vice versa*) would be especially useful, though naturally, it'd be nice to have the full set.

**Comment by 7ombie** [comment]
Hey, @kirawi. Hope you're well.

As I understand #1810, it only permits *selecting* blocks of *equally indented lines*. This issue is about *moving* (jumping) to the *start* of a line, based on its *relationship* to the current line.

For the sake of simplicity and generality, the behavior is inferred from the indentation, but in practice (at a higher level), it allows you to do things like jump to the line that leads the current block (typically an `if`, `while`, `for`, `class` keyword or whatever). It also permits jumping up and down (staying in the same column), while ignoring lines that only contain leading whitespace (similar to the way new cursors skip over deadspace to the next proper line).

**Comment by 7ombie** [comment]
Take this example code (only the indentation is important):

``` coffee
for n in numbers
    x = n * 2
    put n
put numbers
```

If your cursor is on Line 2 somewhere, then #1810 allows you to select Line 2 and Line 3 (because they're at the same level of indentation). This proposal allows you to jump to the start of line 1 (because it's the leading line of the block Line 2 belongs to), and *something like* the opposite. This proposal also allows you to jump from any of the first four columns on Line 1 to the corresponding column on Line 4 (because there's only leading-whitespace in-between), and the *exact* opposite.

**Comment by pascalkuthe** [comment]
Yeah I don't think indentstion based movement is useful compared to tree sitter (especiallyin language which are jot python where indentation is not semantically meaningful). Tree sitter also allows you to select the parent node which can already do want you want (especially if you flip the selection). You can also select the next/previous sibling
